File: Level2/타겟넘버/solution.py
from collections import deque


def solution(numbers, target):
    answer = 0

    queue = deque()
    queue.append((0, 0))

    while queue:
        x, y = queue.popleft()

        if y > len(numbers):
            break
        elif x == target and y == len(numbers):
            answer += 1
        queue.append((x + numbers[y - 1], y + 1))
        queue.append((x - numbers[y - 1], y + 1))

    return answer

File: Level2/타겟넘버/solution.swift
import Foundation

func dfs(_ numbers: [Int], _ target: Int, _ i: Int, _ sum: Int) -> Int {
    if i == numbers.count {
        return sum == target ? 1 : 0
    }

    let a = dfs(numbers, target, i + 1, sum - numbers[i])
    let b = dfs(numbers, target, i + 1, sum + numbers[i])

    return a + b
}

func solution(_ numbers:[Int], _ target:Int) -> Int {
    return dfs(numbers, target, 0, 0)
}